eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/ResourceClientProvider.cs

```diff
@@ -56,6 +56,8 @@ internal static ResourceClientProvider Create(InputModelType model, ResourceMeta
 
         private readonly ResourceMetadata _resourceMetadata;
 
+        internal ResourceMetadata ResourceMetadata => _resourceMetadata;
```

Comment:
I kind of think we should not add this.
Let us try the other way I described in my above comment.